File: expectanim/src/main/java/com/github/florent37/expectanim/core/scale/ScaleAnimExpectationWidth.java
package com.github.florent37.expectanim.core.scale;

import android.view.View;

import com.github.florent37.expectanim.core.position.PositionAnimExpectation;

/**
 * Created by florentchampigny on 20/02/2017.
 */
public class ScaleAnimExpectationWidth extends ScaleAnimExpectation {
    private int width;

    public ScaleAnimExpectationWidth(int width, Integer gravityHorizontal, Integer gravityVertical) {
        super(gravityHorizontal, gravityVertical);
        this.width = width;
    }

    @Override
    public Float getCalculatedValueScaleX(View viewToMove) {
        if(toDp){
            this.width = dpToPx(this.width, viewToMove);
        }

        final int viewToMoveWidth = viewToMove.getWidth();
        if (this.width == 0 || viewToMoveWidth == 0f) {
            return 0f;
        }
        return 1f * this.width / viewToMoveWidth;
    }

    @Override
    public Float getCalculatedValueScaleY(View viewToMove) {
        if(keepRatio){
            return getCalculatedValueScaleX(viewToMove);
        }
        return null;
    }

}

File: expectanim/src/main/java/com/github/florent37/expectanim/core/scale/ScaleAnimExpectationWidth.kt
package com.github.florent37.expectanim.core.scale

import android.view.View

import com.github.florent37.expectanim.core.position.PositionAnimExpectation

/**
 * Created by florentchampigny on 20/02/2017.
 */
class ScaleAnimExpectationWidth(private var width: Int, gravityHorizontal: Int?, gravityVertical: Int?) : ScaleAnimExpectation(gravityHorizontal, gravityVertical) {

    override fun getCalculatedValueScaleX(viewToMove: View): Float? {
        if (toDp) {
            this.width = dpToPx(this.width.toFloat(), viewToMove)
        }

        val viewToMoveWidth = viewToMove.width
        return if (this.width == 0 || viewToMoveWidth.toFloat() == 0f) {
            0f
        } else 1f * this.width / viewToMoveWidth
    }

    override fun getCalculatedValueScaleY(viewToMove: View): Float? {
        return if (keepRatio) {
            getCalculatedValueScaleX(viewToMove)
        } else null
    }

}
